backend/typescript/services/implementations/reviewService.ts

```diff
@@ -234,85 +235,78 @@ class ReviewService implements IReviewService {
+              transaction: txn,
+            });
+          }
         });
 
         // Delete publishers (if necessary)
-        book.publishers.forEach((publisher: PgPublisher) => {
-          PgBookPublisher.findAll({
-            where: { publisher_id: publisher.id },
-          }).then(async (ret: PgBookPublisher[]) => {
-            const bookPublisherIds = ret.map(
-              (bookPublisher: PgBookPublisher) => bookPublisher.book_id,
-            );
-            if (
-              bookPublisherIds.every((bookId) => allBookIds.includes(bookId))
-            ) {
-              // Delete publisher
-              await PgPublisher.destroy({
-                where: { id: [publisher.id] },
-                transaction: txn,
-              });
-            }
+        book.publishers.forEach(async (publisher: PgPublisher) => {
+          const publishersOtherBooks = await PgBookPublisher.findAll({
+            where: {
+              publisher_id: publisher.id,
+              book_id: { [Op.notIn]: allBookIds },
+            },
           });
+          if (publishersOtherBooks.length === 0) {
+            // Delete publisher
+            await PgPublisher.destroy({
+              where: { id: [publisher.id] },
+              transaction: txn,
+            });
+          }
         });
-      }),
-    );
 
-    const deleteBooks = Promise.all(
-      reviewToDelete.books.map(async (book: PgBook) => {
-        // Delete book
-        PgBook.destroy({
-          where: { id: [book.id] },
-          transaction: txn,
-        }).then(() => {
-          // Delete series (if necessary)
-          if (book.series) {
-            PgBook.findAll({
-              where: { series_id: book.series.id },
-            }).then((ret: PgBook[]) => {
-              // We check for 0 because we deleted the book (can't be 1)
-              if (ret.length === 0) {
-                // Delete series
-                PgSeries.destroy({
-                  where: { id: [book.series.id] },
-                  transaction: txn,
-                });
-              }
+        // Delete series (if necessary)
+        if (book.series) {
+          const seriesOtherBooks = await PgBook.findAll({
+            where: {
+              series_id: book.series.id,
+              id: { [Op.notIn]: allBookIds },
+            },
+          });
+          if (seriesOtherBooks.length === 0) {
+            // Delete series
+            await PgSeries.destroy({
+              where: { id: [book.series.id] },
+              transaction: txn,
             });
           }
-        });
+        }
       }),
     );
 
-    const deleteResult = await Promise.all([
-      deletePublishersAndAuthors,
-      deleteBooks,
-    ]).then(() => {
-      return PgReview.destroy({
-        where: { id: [id] },
-        transaction: txn,
-      });
+    await PgReview.destroy({
+      where: { id: [id] },
+      transaction: txn,
     });
 
-    return deleteResult;
+    return true;
```

Comment:
(nit) I think this can just return `void` (unless we capture the return somewhere). An exception should be thrown anyways if something goes wrong

Reply:
The return signals if it is finished, which allows the error to bubble up. This is basically what was there before and it worked so I didn't touch it.